usd/src/sdf_parser/Model.cc

```diff
@@ -145,9 +157,35 @@ namespace usd
         errors.insert(errors.end(), jointErrors.begin(), jointErrors.end());
         return errors;
       }
+
+      if (!markedArticulationRoot && joint.Type() == sdf::JointType::REVOLUTE)
+      {
+        if (!pxr::UsdPhysicsArticulationRootAPI::Apply(modelPrim))
+        {
+          errors.push_back(UsdError(
+                sdf::usd::UsdErrorCode::FAILED_PRIM_API_APPLY,
+                "Unable to mark Xform at path [" + _path +
+                "] as a pxr::UsdPhysicsArticulationRootAPI. "
+                "Some features might not work."));
+        }
+        else
+        {
+          markedArticulationRoot = true;
+        }
+      }
     }
 
-    // TODO(adlarkin) finish parsing model
+    if (!markedArticulationRoot && !_model.Static())
+    {
+      if (!pxr::UsdPhysicsRigidBodyAPI::Apply(modelPrim))
```

Comment:
I have a few questions about this:
1. If we are marking the model as a rigid body API, then why do we need to also mark the link as a rigid body API? I believe that marking only the model as a rigid body API should be sufficient, since the link is a child prim of the model prim.
2. In the first `if` block, why do we check if the model is not static **and** if the model has not been marked as an articulation root? As I understand it, `pxr::UsdPhysicsRigidBodyAPI` only implies static/non-static, which means that articulation root API should not have any influence on whether a prim is marked as a rigid body API or not. The only thing that should influence whether a prim is a rigid body API is whether it is static or not, so I believe the first `if` statement here should only check if the model is not static instead of checking of the model isn't static and also isn't marked as an articulation root API.

Reply:
When I was converting the panda robot to USD I was having some issues. Here it's what I found testing the model in Isaac Sim

 - A `UsdPhysicsRigidBodyAPI` can't be marked as `rigidbody` otherwise I was having issue with the joints
 - If the parent is marked as `UsdPhysicsRigidBodyAPI` then I need to set all children to `rigidbody`.

I don't have a clear explanation about why is this happening but I fix this based on the behaviour in Issac Sim

Reply:
Okay, thanks for the feedback :+1: so if a model prim is marked as an articulation root API, but not as a rigid body API, does it still move without issues? From the initial reading I did on USD, if a prim isn't marked as a rigid body, then I thought this prim would be interpreted as static.